### PR TITLE
Dataset mapping: Revision + handy testing

### PR DESCRIPTION
**Pull request by alpakpinar**
Hey @AndreasAlbert , this PR contains two changes:
1. The dataset merging scheme is split in two: Creation of the mapping is moved into one function and the application of it is moved to another. 
2. A script to quickly test the current form of merging is added. This gets the mapping for all datasets listed in our v7 files, and prints them out in a rather readable way. The printout looks something like this:
image

I can play with the output format if you like, or make some changes like instead of printing out, save it into a log file, what do you think? Thanks!

### CODE REVIEW

File: bucoffea/plot/util.py [hunk 1]
```diff
@@ -266,6 +262,21 @@ def merge_datasets(histogram):
...
+    :type histogram: Coffea histogram
+    :return: Modified histogram
+    :rtype: Coffea histogram
+    """
+    all_datasets = list(map(str, histogram.identifiers('dataset')))
+    # Create the mapping for the datasets in this histogram
+    mapping = create_dataset_mapping(all_datasets)
+    
     # Remove empty lists
     tmp = {}
```
Comment (AndreasAlbert) [inline bucoffea/plot/util.py]: Why did you choose to keep this cleanup code in merge_datasets? wouldnt it make sense to also move it to create_dataset_mapping?

**Comment by AndreasAlbert** [comment]
thanks @alpakpinar , this looks great! Just the one question above

File: bucoffea/plot/util.py (hunk 1 above)
Comment (alpakpinar) [reply bucoffea/plot/util.py]: Ah you're right, it would make more sense to move it to create_dataset_mapping, I'll make that change soon

**Comment by alpakpinar** [comment]
Hey @AndreasAlbert , I now moved the code which cleans the empty entries to create_dataset_mapping function. I kept the other piece of "post-proc" code in merge_datasets func though, so that create_dataset_mapping handles the mapping for the datasets we are catching with regex, and merge_datasets handles the rest (I guess its just one-to-one mapping anyway). If you want to do so, I can also move that second piece of code to create_dataset_mapping as well, what do you think, thanks!

**Comment by AndreasAlbert** [comment]
thanks for the update. let's move that second step into the separate function as well. To me, the idea here is that the creation of the mapping and its application are completely separated.

**Comment by alpakpinar** [comment]
Ah yes, this actually makes more sense. I'll make the changes shortly, thanks!

**Comment by alpakpinar** [comment]
Hey @AndreasAlbert , now that piece of code is moved into the helper function as well. Let me know if you need anything else, thank you!

**Comment by AndreasAlbert** [comment]
thanks @alpakpinar !

**Pull request merged**